File: app-android/app/src/main/java/com/health2u/data/local/entity/HealthInsightEntity.kt
package com.health2u.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey
import androidx.room.Index

@Entity(
    tableName = "health_insights",
    indices = [Index(value = ["userId"]), Index(value = ["timestamp"])]
)
data class HealthInsightEntity(
    @PrimaryKey val id: String,
    val userId: String,
    val type: String,
    val title: String,
    val description: String,
    val metricValue: Double?,
    val timestamp: Long,
    val createdAt: Long
)

File: app-ios/Sources/Health2u/Data/Local/HealthInsightEntity.swift
import Foundation

public struct HealthInsightEntity: Sendable, Equatable, Identifiable {
    public let id: String
    public var userId: String
    public var type: String
    public var title: String
    public var descriptionText: String
    public var metricValue: Double?
    public var timestamp: Date
    public var createdAt: Date

    public init(
        id: String,
        userId: String,
        type: String,
        title: String,
        descriptionText: String,
        metricValue: Double? = nil,
        timestamp: Date,
        createdAt: Date
    ) {
        self.id = id
        self.userId = userId
        self.type = type
        self.title = title
        self.descriptionText = descriptionText
        self.metricValue = metricValue
        self.timestamp = timestamp
        self.createdAt = createdAt
    }
}
